content/kotlin/concepts/enums/enums.md

```diff
@@ -0,0 +1,226 @@
+    DIESEL,
+    PREMIUM,
+    UNLEADED
+}
+fun main()
+{
+    // A simple demonstration of properties
+    for (gasoline in GASOLINE.values()) {println("${gasoline.ordinal} = ${gasoline.name}")}}
+```
+
+This prints the following output:
+
+```shell
+0 = DIESEL
+1 = PREMIUM
+2 = UNLEADED
+```
+
+and the use of methods:
+
+```kotlin
+enum class GASOLINE {
+    DIESEL,
+    PREMIUM,
+    UNLEADED
+}
+fun main()
+{
+    // A simple demonstration of methods
+    for (gasoline in GASOLINE.values()) println("${GASOLINE.valueOf("UNLEADED")}")
+}
+```
+
+the output:
+
+```shell
+UNLEADED
+UNLEADED
+UNLEADED
+```
+The use of when expression -
+
+One of the interesting features of kotlin enums is the complete elimination of the need to use the else clause because of the restriction on the value that a type can take.
+
+## Example
+
+```kotlin
+enum class TRADED_METALS{
+    GOLD,
+    SILVER,
+    COPPER,
+    IRON,
+    TITANIUM,
+    PLATINUM,
+    LITHIUM;
+}
+   
+fun main(){
+    when(TRADED_METALS.IRON){
+        TRADED_METALS.GOLD -> println("The symbol of Gold is Au")
+        TRADED_METALS.SILVER -> println("The symbol of Silver is Ag")
+        TRADED_METALS.COPPER -> println("The symbol of Copper is Cu")
+        TRADED_METALS.IRON -> println("The symbol of Iron is Fe")
+        TRADED_METALS.TITANIUM -> println("The symbol of Titanium is Ti")
+        TRADED_METALS.PLATINUM -> println("The symbol of Platinum is Pt")
+        TRADED_METALS.LITHIUM -> println("The symbol of Lithium is Li")
+        // Adding an else clause will generate a warning
+    }
+}
+```
+
+the output:
+
+```shell
+The symbol of Iron is Fe
+```
+
+Use of constructors to add more properties -
+
+## Example
+
+```kotlin
+enum class COLORS(val symbol: String) {
+    RED(symbol: "war"),
+    BLUE(symbol: "peace"),
+    WHITE(ymbol: "neutrality")
+}
+fun main()
+{
+    printIn(COLORS.RED.symbol)
+}
+```
+the output:
+
+```shell
+war
+```
+
+Implementing interface - 
+
+## Example
+
+```kotlin
+interface IAccountInterestRates {
+    fun getInterestRate() : Double
+}
+ 
+enum class AccountTypes(val minBalance: Int) : IAccountInterestRates {
+    SAVINGS_ACCOUNT(0) {
+        override fun getInterestRate(): Double {
+            return(1.2)
+        }
+    },
+    CHECKING_ACCOUNT(100) {
+        override fun getInterestRate(): Double {
+            return(0.5)
+        }
+    },
+    WALLET(100) {
+        override fun getInterestRate(): Double {
+            return(0.0)
+        }
+    }
+}
+ 
+fun main() {
+    // calling static method in Enum
+    println("Savings bank interest rate is : ${AccountTypes.SAVINGS_ACCOUNT.getInterestRate()}")
+}
+```
+
+output is:
+
+```shell
+Savings bank interest rate is : 1.2
+```
+
+Enums as Anonymous Classes –
+
+Enum constants also behaves as anonymous classes by implementing their own functions along with overriding the abstract functions of the class. The most important thing is that each enum constant must be override.
```

Comment:
[nitpick/question] “What should enum constants be overridden by?” is something developers would ask as they read this paragraph. Consider adding the information that I provided below to your paragraph.
```suggestion
Enum constants behave as anonymous classes by implementing their own functions along with overriding the abstract functions of the class. The most important thing is that each enum constant must be overridden by using its own functions along with overriding the abstract functions of the class.
```